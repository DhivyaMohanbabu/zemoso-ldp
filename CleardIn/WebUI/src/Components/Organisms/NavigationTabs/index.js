import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Grid, makeStyles } from '@material-ui/core';
import Tab from '../../Atoms/Tab';
import NewRow from '../../Molecules/NewEntry';

const useStyles = makeStyles({
  tab: {
    marginRight: '18px',
    marginTop: '2px',
  },
  newTab: {
    marginLeft: '15px',
    marginTop: '2px',
  },
});

const NavigationTabs = ({ tabs }) => {
  const classes = useStyles();
  const [activeTab, setActiveTab] = useState(0);
  return (
    <Grid container direction="row" justify="flex-start">
      {tabs && tabs.map((tab, index) => (
        <Grid item onClick={() => setActiveTab(index)} key={index} className={classes.tab}>
          <Tab title={tab} isSelected={index === activeTab} />
        </Grid>
      ))}
      <Grid item className={classes.newTab}>
        <NewRow text="New Tab" />
      </Grid>
    </Grid>
  );
};

NavigationTabs.propTypes = {
  tabs: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default NavigationTabs;
