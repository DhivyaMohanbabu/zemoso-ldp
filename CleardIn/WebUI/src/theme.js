import { grey } from '@material-ui/core/colors';
import { createMuiTheme } from '@material-ui/core/styles';
// import 'typeface-roboto';

const theme = createMuiTheme({
  palette: {
    primary: {
      main: '#1665d8',
      light: '#b0f1f0',
    },
    secondary: {
      main: '#9ea0a5',
      dark: grey,
    },
    error: {
      main: '#ffada5',

    },
    success: {
      main: '#daefa9',
    },
    info: {
      main: grey[900],
      light: '#fdd570',
      dark: grey[600],
    },
    text: {
      primary: '#1c1c21', // '#3e3f42',
      secondary: '#5c5c5c', // #6b6c6f',
    },
    grey: {
      100: '#f8fafb',
      200: '#f2f4f6',
      300: '#e8ecef',
      400: '#98a9bc',
      500: '#778ca2',
    },
    clearedIn: {
      grey: {
        100: '#f6f6f6',
        200: '#eeeeee',
        300: '#cccccc',
        400: '#949494',
        500: '#666666',
      },
      selectedElementBGColor: '#eaf0ff', // rgba(77, 124, 254, 0.12),
      dark: '#5c5c5c',
      bodyBackground: '#fafafa',
      blue: '#1d4cd7',
    },
    relationShip: {
      vendor: '#daefa9',
      customer: '#fdd570',
      partner: '#b0f1f0',
    },
  },
  typography: {
    fontFamily: ['Roboto', 'Rubik', 'sans-serif'].join(','),
    h1: {
      fontSize: '1.875rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    h2: {
      fontSize: '1.625rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    h3: {
      fontSize: '1.25rem',
      fontWeight: 300,
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    h4: {
      fontSize: '1.125rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    h5: {
      fontSize: '1rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 1.5,
      letterSpacing: 'normal',
    },
    h6: {
      fontSize: '1.25rem',
      fontWeight: 500,
      fontStyle: 'normal',
      lineHeight: 1.05,
      letterSpacing: 'normal',
    },
    subtitle1: {
      fontSize: '0.938rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    subtitle2: {
      fontSize: '0.938rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    caption: {
      fontSize: '0.875rem',
      fontWeight: 500,
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    link: {
      fontSize: '0.875rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 1.5,
      letterSpacing: 'normal',
    },
    body1: {
      fontSize: '0.875rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 1.14,
      letterSpacing: 'normal',
    },
    body2: {
      fontFamily: 'Rubik',
      fontSize: '0.875rem',
      fontWeight: 300,
      fontStyle: 'normal',
      lineHeight: 1.5,
      letterSpacing: 'normal',
    },
    button: {
      fontSize: '0.75rem',
      fontWeight: 500,
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
      textTransform: 'none',
    },
  },
  overrides: {
    MuiButton: {
      root: {
        maxWidth: '200px',
        display: 'block',
      },
      label: {
        whiteSpace: 'nowrap',
        overflow: 'hidden',
        textOverflow: 'ellipsis',
      },
    },
  },
});

export default theme;
