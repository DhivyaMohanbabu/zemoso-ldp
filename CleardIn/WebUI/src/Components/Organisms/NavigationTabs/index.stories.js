import React from 'react';
import NavigationTabs from '.';

export default {
  title: 'organism/NavigationTabs',
  component: NavigationTabs,
};

const Template = ({ ...args }) => <NavigationTabs {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  tabs: ['Top 100', 'Top 1000', 'External partners', 'Box communication base'],
};
