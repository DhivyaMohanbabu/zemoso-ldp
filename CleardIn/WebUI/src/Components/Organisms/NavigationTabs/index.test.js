import React from 'react';
import '@testing-library/jest-dom';
import { fireEvent, render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/core/styles';
import NavigationTabs from '.';
import theme from '../../../theme';

const allProps = {
  tabs: ['Top 100', 'Top 1000', 'External partners', 'Box communication base'],
};

it('renders NavigationTabs snapshot correctly', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <NavigationTabs {...allProps} />
    </ThemeProvider>,
  );
  expect(container).toMatchSnapshot();
});

it('NavigationTabs onClick tab', () => {
  const { getByText } = render(
    <ThemeProvider theme={theme}>
      <NavigationTabs {...allProps} />
    </ThemeProvider>,
  );
  fireEvent.click(getByText(allProps.tabs[1]));
  expect(getByText(allProps.tabs[1]).parentElement).toMatchSnapshot();
});
