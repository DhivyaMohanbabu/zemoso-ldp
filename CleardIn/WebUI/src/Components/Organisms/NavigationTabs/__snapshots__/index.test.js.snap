// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationTabs onClick tab 1`] = `
<div
  class="makeStyles-tabSpan-10 makeStyles-currentTab-11"
>
  <p
    class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap"
  >
    Top 1000
  </p>
</div>
`;

exports[`renders NavigationTabs snapshot correctly 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container"
  >
    <div
      class="MuiGrid-root makeStyles-tab-1 MuiGrid-item"
    >
      <div
        class="makeStyles-tabSpan-3 makeStyles-currentTab-4"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap"
        >
          Top 100
        </p>
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-tab-1 MuiGrid-item"
    >
      <div
        class="makeStyles-tabSpan-3 "
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap"
        >
          Top 1000
        </p>
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-tab-1 MuiGrid-item"
    >
      <div
        class="makeStyles-tabSpan-3 "
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap"
        >
          External partners
        </p>
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-tab-1 MuiGrid-item"
    >
      <div
        class="makeStyles-tabSpan-3 "
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap"
        >
          Box communication base
        </p>
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-newTab-2 MuiGrid-item"
    >
      <div
        class="makeStyles-root-5"
      >
        <div
          class="makeStyles-Icon-6"
        >
          <span
            aria-hidden="true"
            class="material-icons MuiIcon-root"
          >
            +
          </span>
        </div>
        <div
          class="makeStyles-text-7"
        >
          <body1
            class="MuiTypography-root MuiTypography-body1"
            data-testid="text"
          >
            New Tab
          </body1>
        </div>
      </div>
    </div>
  </div>
</div>
`;
